### PR TITLE
fix find parent node issue

### LINKED ISSUE #22342: node.getParent API doesn't return the parent node for table column node.
when the node is a table column, in the context menu command handler, 
```
const node = await azdata.objectexplorer.getNode(context.connectionProfile.id, context.nodeInfo.nodePath);
const parentNode = await node?.getParent();
```

the parentNode here is undefined. for other node types, this works fine.

### PR DESCRIPTION
**Pull request by alanrenmsft**
This PR fixes #22342

Depends on: https://github.com/microsoft/sqltoolsservice/pull/1949

Root cause:
the schema information is set on the table column node, but the node path doesn't have it, as a result, we couldn't find the parent path. 

Fix:
The extension side shouldn't have knowledge of how the provider is constructing the node path (especially schema is not a common concept across all providers), if we need to know it, we should get it directly from the providers. 

node-path

this is the actual change: https://github.com/microsoft/azuredatastudio/pull/22356/files#diff-29942eaeb9ec5e20de436b4fd36ae4ee41588c3d50e58761787ccc863ec50a4aR100, all other changes are just to add the new property.

STS changes:
image

### CODE REVIEW

**Comment by alanrenmsft** [comment]
this is the actual change: https://github.com/microsoft/azuredatastudio/pull/22356/files#diff-29942eaeb9ec5e20de436b4fd36ae4ee41588c3d50e58761787ccc863ec50a4aR100, all other changes are just to add the new property.

File: src/sql/workbench/api/common/extHostObjectExplorer.ts [hunk 1]
```diff
@@ -96,20 +97,29 @@ export class ExtHostObjectExplorerNode implements azdata.objectexplorer.ObjectEx
...
+			// -1 to remove the / as well
+			let parentPathEndIndex: number = this.nodePath.lastIndexOf(nodePathName) - 1;
+			parentPath = this.nodePath.slice(0, parentPathEndIndex)
+		} else {
+			parentPath = this.parentNodePath;
+		}
+
+		if (!parentPath) {
 			// At root node
 			return Promise.resolve(undefined);
```
Comment (Charles-Gagnon) [inline src/sql/workbench/api/common/extHostObjectExplorer.ts:120]: nit: Update return type of this function to add `| undefined`

**Review by Charles-Gagnon — APPROVED** [review]

**Comment by Charles-Gagnon** [comment]
You do have a compile error to fix it looks like though

**Comment by alanrenmsft** [comment]
@Charles-Gagnon yes, I am working on fixing them. they didn't show up locally within the watch window, I had to rerun it for the errors to appear.

**Review by kburtram — APPROVED** [review]

File: src/sql/workbench/api/common/extHostObjectExplorer.ts (hunk 1 above)
Comment (alanrenmsft) [reply src/sql/workbench/api/common/extHostObjectExplorer.ts:120]: done

**Pull request merged**